src/wazuh_db/wdb_integrity.c

```diff
@@ -520,14 +579,33 @@ int wdbi_check_sync_status(wdb_t *wdb, wdb_component_t component) {
 
     cJSON* j_last_attempt = cJSON_GetObjectItem(j_sync_info->child, "last_attempt");
     cJSON* j_last_completion = cJSON_GetObjectItem(j_sync_info->child, "last_completion");
+    cJSON* j_checksum = cJSON_GetObjectItem(j_sync_info->child, "last_global_checksum");
 
-    if ( cJSON_IsNumber(j_last_attempt) && cJSON_IsNumber(j_last_completion)) {
+    if ( cJSON_IsNumber(j_last_attempt) && cJSON_IsNumber(j_last_completion) && cJSON_IsString(j_checksum)) {
         int last_attempt = j_last_attempt->valueint;
         int last_completion = j_last_completion->valueint;
 
-        // Return 0 if there was not at least one successful syncronization or if the syncronization is in process
-        result = (last_completion != 0 && last_attempt <= last_completion );
+        // Return 0 if there was not at least one successful syncronization or
+        // if the syncronization is in process or there was an error iun the syncronization
+        if (result = (last_completion != 0 && last_attempt <= last_completion ), result) {
+            // Verifying the integrity checksum
+            os_sha1 hexdigest;
+            char *checksum = NULL;
+
+            switch (wdbi_checksum(wdb, component, hexdigest)) {
+            case -1:
+                result = OS_INVALID;
+                break;
+
+            case 0:
+                result = 0;
```

Comment:
An empty table is a possible and valid state, but we are considering that state as "not synced"